beacon-chain/state/stateutil/trie_helpers_test.go

```diff
@@ -27,6 +28,57 @@ func TestReturnTrieLayer_OK(t *testing.T) {
 	assert.NoError(t, err)
 	newRoot := *layers[len(layers)-1][0]
 	assert.Equal(t, root, newRoot)
+
+	flags := &features.Flags{}
+	flags.EnableVectorizedHTR = true
+	reset := features.InitWithReset(flags)
+	defer reset()
+
+	layers, err = stateutil.ReturnTrieLayer(roots, uint64(len(roots)))
+	assert.NoError(t, err)
+	lastRoot := *layers[len(layers)-1][0]
+	assert.Equal(t, root, lastRoot)
+}
+
+func BenchmarkReturnTrieLayer_NormalAlgorithm(b *testing.B) {
+	newState, _ := util.DeterministicGenesisState(b, 32)
+	root, err := stateutil.RootsArrayHashTreeRoot(newState.BlockRoots(), uint64(params.BeaconConfig().SlotsPerHistoricalRoot), "BlockRoots")
+	require.NoError(b, err)
+	blockRts := newState.BlockRoots()
+	roots := make([][32]byte, 0, len(blockRts))
+	for _, rt := range blockRts {
+		roots = append(roots, bytesutil.ToBytes32(rt))
+	}
```

Comment:
Perhaps make this into it's own helper function? it appears a few times in this file